scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoInfer.scala

```diff
@@ -1,25 +1,39 @@
 package scalafix.internal.rule
 
+import metaconfig.{Conf, Configured}
+
 import scala.meta._
+import scalafix.internal.config.NoInferConfig
 import scalafix.lint.LintCategory
 import scalafix.lint.LintMessage
-import scalafix.rule.RuleCtx
-import scalafix.rule.SemanticRule
+import scalafix.rule.{Rule, RuleCtx, SemanticRule}
 import scalafix.util.SemanticdbIndex
 import scalafix.util.SymbolMatcher
 
-case class NoInfer(index: SemanticdbIndex)
+case class NoInfer(index: SemanticdbIndex,
+                   config: NoInferConfig = NoInferConfig.default)
     extends SemanticRule(index, "NoInfer")
     with Product {
-  private val badSymbol = SymbolMatcher.exact(NoInfer.badSymbols: _*)
+  private val badSymbol = SymbolMatcher.exact(NoInfer.badSymbols ++ config.extraSymbols: _*)
+
+  def this(index: SemanticdbIndex) =
+    this(index, NoInferConfig.default)
+
+  override def init(config: Conf): Configured[Rule] =
+    config
+      .getOrElse("noInfer")(NoInferConfig.default)
+      .map(c => NoInfer(index, c))
+
   val error: LintCategory =
     LintCategory.error(
       """The Scala compiler sometimes infers a too generic type such as Any.
         |If this is intended behavior, then the type should be explicitly type
         |annotated in the source.""".stripMargin
     )
-  override def check(ctx: RuleCtx): Seq[LintMessage] =
-    ctx.index.synthetics.flatMap {
+
+  override def check(ctx: RuleCtx): Seq[LintMessage] = {
+
```

Comment:
Is this change necessary?

Reply:
No. I didn't undo enough when backtracking a change.